src/os/bluestore/BlueFS.cc

```diff
@@ -1075,6 +1081,20 @@ int BlueFS::mount()
         shared_alloc->bluefs_used += q.length;
         alloc[q.bdev]->init_rm_free(q.offset, q.length);
       } else if (!is_shared) {
+        if (locked_alloc[q.bdev].length) {
+          auto locked_offs = locked_alloc[q.bdev].offset;
+          if (q.offset <= locked_offs && q.offset + q.length > locked_offs) {
```

Comment:
The condition is wrong.
We need to take action if `q` has a non-zero intersection with `locked_alloc`.
I guess we can silently assume `q.offset + q.length <= locked.offset + locked.length`.
But if `locked.offset < q. offset < q.offset + q.length <= locked.offset + locked.length` ; we still want to take action.

So, the condition should be `q.offset + q.length > locked.offset`

Reply:
Reworked in a different way, should be fine now